src/test/java/selenium/model/generator/UserCredentialsFeatureGenerator.java

```diff
@@ -0,0 +1,40 @@
+package selenium.model.generator;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import selenium.model.UserCredentials;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class UserCredentialsFeatureGenerator implements FeatureGenerator {
+    private final static Logger LOG = LogManager.getLogger();
+    private final List<UserCredentials> userCredentials;
+
+    public UserCredentialsFeatureGenerator(List<UserCredentials> userCredentials) {
+        this.userCredentials = userCredentials;
+    }
+
+    @Override
+    public void generateFeatureFile(Path pathToFeatureTemplate, Path pathToFeatureOutput) {
+        try {
+            generateCsvDrivenFeatureOutlineFile(pathToFeatureTemplate, pathToFeatureOutput);
+        } catch (IOException e) {
+            LOG.error("Cannot generate output feature file.", e);
+        }
+    }
+
+    private void generateCsvDrivenFeatureOutlineFile(Path pathToFeatureTemplate, Path pathToFeatureOutput)
+            throws IOException {
+        Files.delete(pathToFeatureOutput);
+        Files.createFile(pathToFeatureOutput);
+
+        List<String> templateLines = Files.readAllLines(pathToFeatureTemplate);
```

Comment:
If you deleted the file in line 31 that you are reading here, then it will be an empty list. You should either not delete it (if it is needed), or should not read it afterwards (and then use collect method in line 35)

Reply:
It is deleted so it can be generated again from the input data. 
Otherwise it would append the new lines to the previous ones.

Reply:
OK